docs/csharp/language-reference/compiler-messages/cs8411.md

```diff
@@ -0,0 +1,34 @@
+---
+title: Compiler Error CS8411
+ms.date: 07/11/2020
+f1_keywords:
+  - "CS8411"
+helpviewer_keywords:
+  - "CS8411"
+author: Youssef1313
+---
+# Compiler Error CS8411
+
+Asynchronous foreach statement cannot operate on variables of type 'type' because 'type' does not contain a suitable public instance definition for 'GetAsyncEnumerator'
+
+`await foreach` statement operates only on types having a definition of `GetAsyncEnumerator`, such as `IAsyncEnumerable<T>`.
+
+## To correct this error
+
+Replace `await foreach` with `foreach`.
+
+## Example
+
+<!-- TODO: The current example is a little bit poor. -->
+
+```csharp
+using System.Collections.Generic;
+using System.Threading.Tasks;
+public class C
+{
+    async Task M(IAsyncEnumerator<int> enumerator)
+    {
+        await foreach (int i in enumerator) { }
+    }
+}
+```
```

Comment:
```suggestion
```csharp
using System.Collections.Generic;
using System.Threading.Tasks;

class Program
{
    async Task Main(IAsyncEnumerator<int> enumerator)
    {
        // error CS8411: Asynchronous foreach statement cannot operate on variables 
        // of type 'IAsyncEnumerator<T>' because 'IAsyncEnumerator<T>' does not 
        // contain a suitable public instance definition for 'GetAsyncEnumerator'
        await foreach (int i in enumerator) 
        {
        }
    }
}
```
```